presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -447,6 +448,28 @@ private void processNewSources()
         }
     }
 
+    private void addBlockedOperator(List<Operator> blockedOperators, List<ListenableFuture<?>> blockedFutures, Operator operator, ListenableFuture<?> blockedFuture)
+    {
+        if (!blockedOperators.isEmpty() && getLast(blockedOperators) == operator) {
```

Comment:
explain for in comment why we need this.

Reply:
will do